src/child/actions/window.js

```diff
@@ -57,30 +57,104 @@ export function resizeError() {
     };
 }
 
-export function resizeToCompact() {
+export function updatingOptions() {
+    return {
+        type: ACTION_TYPES.UPDATING_OPTIONS
+    };
+}
+
+export function updatingOptionsSuccess() {
+    return {
+        type: ACTION_TYPES.OPTIONS_SUCCESS
+    };
+}
+
+export function updatingOptionsError() {
+    return {
+        type: ACTION_TYPES.OPTIONS_ERROR
+    };
+}
+
+function updateOptionsToCompact() {
+    return dispatch => {
+        dispatch(updatingOptions());
+        const [compactWindowWidth, compactWindowHeight] = configService.getCompactWindowDimensions();
```

Comment:
Could name these `minWidth` and `minHeight` to use property value shorthand in the object passed to `updateOptions()` below - both in `updateOptionsToCompact()` and `updateOptionsToDefault()`.

(Sorry - my fault for messing with the code example in the comment I made!)